src/App.jsx

```diff
@@ -1,65 +1,136 @@
-          <animateTransform
-            attributeName="transform"
-            type="rotate"
-            from="0 12 12"
-            to="360 12 12"
-            dur="1s"
-            repeatCount="indefinite"
-          />
-        </path>
-      </svg>
+  const copyToClipboard = (svg) => {
+    copyText(ReactDOMServer.renderToStaticMarkup(svg));
+  };
+  const searchIconsByName = (name, theme, option) => {
+    const iconDivs = [];
+    // eslint-disable-next-line no-plusplus
+    for (let i = 0; i < EOSIconsList[option].length; i++) {
+      const icon = EOSIconsList[option][i];
+      const isFilledAvailable = "hasOutlined" in icon && icon.hasOutlined;
+      const isFilledSelected = theme === "Filled";
+      if (
+        icon.name.indexOf(name) !== -1 &&
+        (!isFilledSelected || (isFilledSelected && isFilledAvailable))
+      ) {
+        const nameIcon = `Eos${icon.name}${
+          isFilledSelected ? "Filled" : "Outlined"
+        }`;
+        const EOSReactIcon = EOSIcons[nameIcon];
+        if (EOSReactIcon === undefined) {
+          // eslint-disable-next-line no-continue
+          continue;
+        }
+        iconDivs.push(
+          <div
+            className="image-container"
+            key={nameIcon}
+            onClick={() => {
+              copyToClipboard(EOSReactIcon({ size: "xxxl" }));
+            }}
+          >
+            {EOSReactIcon({ size: "xl" })}
+          </div>
+        );
+      }
+    }
+    if (iconDivs.length === 0) {
+      return null;
+    }
+    return (
+      <div key={option}>
+        <div className="category-container">
+          <span>{option}</span>
+        </div>
+        <div className="image-box">{iconDivs}</div>
+      </div>
     );
-    request.onload = () => {
-      if (!request.response.includes("Couldn't find")) {
-        updateSearchGif(<div />);
-        updateFile(request.response);
-        updatePlugin(request.response);
-      } else {
-        updateSearchGif(<div />);
-        updatePlugin(<p>Svg not found. Try searching for abstract.</p>);
+  };
+  const createIcons = (option) => {
+    const iconDivs = [];
+    const limit =
+      EOSIconsList[option].length < 10 ? EOSIconsList[option].length : 10;
+    // eslint-disable-next-line no-plusplus
+    for (let i = 0; i < limit; i++) {
+      const icon = EOSIconsList[option][i];
+      const name = EOSIcons[`Eos${icon.name}Filled`]
+        ? `Eos${icon.name}Filled`
+        : `Eos${icon.name}Outlined`;
+      const EOSReactIcon = EOSIcons[name];
+      if (EOSReactIcon !== undefined) {
+        iconDivs.push(
+          <div
+            className="image-container"
+            key={name}
+            onClick={() => {
+              copyToClipboard(EOSReactIcon({ size: "xxxl" }));
+            }}
+          >
+            {EOSReactIcon({ size: "xl" })}
+          </div>
+        );
       }
-    };
-    request.send();
+    }
+    return (
+      <div key={option}>
```

Comment:
this code is the same as the one in line 64. Why not have the view of this in a component with props that we can reuse? same for the code in line 84 which is also (almost) repeated in line 48 (only difference is a different variable passed..but with a prop this duplication would be removed)